docs/DeveloperGuide.adoc

```diff
@@ -297,69 +312,430 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
++
+[none]
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for the correct data.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: UC03 - Edit module
+
+*<<mss, MSS>>*
+
+1.  User requests to edit a module.
+2.  User provides the module code.
+3.  User provides the new description.
+4.  Mod Manager edits the module.
++
+Use case ends.
+
+*<<extensions, Extensions>>*
+
+[none]
+* 2a. The given module code is invalid.
++
+[none]
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for the correct data.
++
+Use case resumes at step 2.
+
+* 3a. The new description is more than 60 characters.
++
+[none]
+** 3a1. Mod Manager shows an error message.
+** 3a2. Mod Manager requests for the correct data.
++
+Use case resumes at step 3.
 
 [discrete]
-=== Use case: Delete person
+=== Use case: UC04 - Delete module
 
-*MSS*
+*<<mss, MSS>>*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to delete a module.
+2.  User provides the module code.
+3.  Mod Manager deletes the module.
 +
 Use case ends.
 
-*Extensions*
+*<<extensions, Extensions>>*
+
+[none]
+* 2a. The given module code is invalid.
++
+[none]
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for the correct data.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: UC05 - Add class
+*<<mss, MSS>>*
+
+1. User request to add a class.
+2. User provides the details of the class.
+3. Mod Manager adds a class.
++
+Use case ends.
 
+*<<extensions, Extensions>>*
 [none]
-* 2a. The list is empty.
+* 2a. Compulsory fields are not provided or none of the optional fields provided.
++
+[none]
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for the correct data.
++
+Use case resumes at step 2.
+
+[discrete]
+=== Use case: UC06 - Edit class
+*<<mss, MSS>>*
+
+1. User request to edit a class.
+2. User provides the module code and class type of the class to be edited.
+3. User provides the new details of the class.
+4. Mod Manager edits the class.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+*<<extensions, Extensions>>*
+[none]
+* 2a. Compulsory fields are not provided or fields provided are not valid.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for valid data.
 +
 Use case resumes at step 2.
 
-_{More to be added}_
+[discrete]
+=== Use case: UC07 - Delete class
+*<<mss, MSS>>*
+
+1. User requests to delete a class.
+2. User provides module code and class type.
+3. Mod Manager deletes the class.
++
+Use case ends.
+
+*<<extensions, Extensions>>*
+[none]
+* 2a. Module code or class type provided is invalid.
++
+[none]
+** 2a1. Mod Manager shows an error message.
+** 2a2. Mod Manager requests for valid data.
++
+Use case resumes at step 2.
```

Comment:
I think you can say "Steps 2a1 - 2a2  are repeated until the data entered is correct" before the use case resumes :)